src/OpenSpaceToolkit/Astrodynamics/Dynamics/Tabulated.cpp

```diff
@@ -190,6 +188,35 @@ VectorXd Tabulated::computeContribution(
         contribution(i) = interpolators_[i]->evaluate(epoch);
     }
 
+    // Transform the contribution to the desired frame if necessary
+    if (frameSPtr_ != aFrameSPtr)
+    {
+        if (!frameSPtr_->isQuasiInertial() && aFrameSPtr->isQuasiInertial())
+        {
+            // TBM: Replace with an actual proper logging mechanism
+            std::cout << "WARNING: This TabulatedDynamics has its contributions stored in a non quasi-inertial frame, "
+                         "but they are being requested in a quasi-inertial frame. Ensure that the trajectory "
+                         "used to generate the contributions with this TabulatedDynamics in the first place "
+                         "is still valid, otherwise this may lead to inaccurate results."
+                         "inaccurate results."
+                      << std::endl;
+        }
+
+        const Shared<CoordinateBroker> coordinatesBrokerSPtr = std::make_shared<CoordinateBroker>();
+
+        Index coordinateSubsetIndex = 0;
+        for (const auto& coordinateSubset : writeCoordinateSubsets_)
+        {
+            coordinateSubsetIndex = coordinatesBrokerSPtr->addSubset(coordinateSubset);
+
+            if (coordinateSubset->getSize() == 3)
+            {
+                contribution.segment(coordinateSubsetIndex, coordinateSubsetIndex + 3) =
+                    coordinateSubset->inFrame(anInstant, contribution, frameSPtr_, aFrameSPtr, coordinatesBrokerSPtr);
+            }
```

Comment:
```suggestion
            contribution.segment(coordinateSubsetIndex, coordinateSubsetIndex + 3) =
                coordinateSubset->inFrame(anInstant, contribution, frameSPtr_, aFrameSPtr, coordinatesBrokerSPtr);
```
What's the reason we're only applying this to size 3?
IMO if it's a size 1 it can just return the unconverted scalar value that way we don't have to go around checking.

Reply:
@vishwa2710 because I think there may be an issue with the `CoordinateSubset` frame conversion.
[See my comment](https://github.com/open-space-collective/open-space-toolkit-astrodynamics/pull/373#discussion_r1553478373)

What do you think?